packages/jira-adapter/test/change_validators/field_contexts/custom_field_with_10K_options.test.ts

```diff
@@ -0,0 +1,56 @@
+/*
+*                      Copyright 2023 Salto Labs Ltd.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with
+* the License.  You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import { CORE_ANNOTATIONS, ElemID, InstanceElement, ObjectType, ReferenceExpression, Values, toChange } from '@salto-io/adapter-api'
+import { JIRA } from '../../../src/constants'
+import { customFieldsWith10KOptionValidator } from '../../../src/change_validators/field_contexts/custom_field_with_10K_options'
+
+const generateOptions = (count: number): Values =>
+  Array.from({ length: count }, (_, i) => ({
+    [`p${i}`]: {
+      value: `p${i}`,
+      disabled: false,
+      position: i,
+    },
+  })).reduce((acc, option) => ({ ...acc, ...option }), {})
```

Comment:
```suggestion
  Object.fromEntries(_.range(count).map(i => [[
    `p${i}`,
     {
      value: `p${i}`,
      disabled: false,
      position: i,
    },
   ]]))
```
I think this is faster since you don't copy the object on each option